tests/ui/pattern/issue-117626.rs

```diff
@@ -0,0 +1,21 @@
+// check-pass
+
+#[derive(PartialEq)]
+struct NonMatchable;
+
+impl Eq for NonMatchable {}
+
+#[derive(PartialEq, Eq)]
+enum Foo {
+    A(NonMatchable),
+    B(*const u8),
+}
+
+const CONST: Foo = Foo::B(std::ptr::null());
```

Comment:
@compiler-errors The goal of ﻿`is_ty_included_by_const_val` is to determine if the violation type is incorporated within the current `ConstVal` type.

In this scenario, the type of ﻿`CONST` is `﻿Foo`, but the type is `﻿RawPtr` after evaluation, which does not include `﻿NonMatchable`. Therefore, ﻿`is_ty_included_by_const_val` should return ﻿false.